version: '3.9'

services:
  blue_bot:
    container_name: blue_bot_container
    image: blue_bot_image
    build:
      context: bot
      dockerfile: Dockerfile
    env_file:
      - ./.env
    depends_on:
      - blue_django

  blue_django:
    container_name: blue_django_container
    image: django_image
    build: django_blue/
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - ./django_blue/:/usr/src/django_blue
    ports:
      - '8000:8000'
    env_file:
      - .env
    depends_on:
      - blue_bot_db

  blue_bot_db:
    container_name: ${DB_HOST}
    image: postgres:alpine3.17
    volumes:
      - ./create_table.sql:/docker-entrypoint-initdb.d/create_table.sql
      - postgres_data:/var/lib/postgresql/data

    ports:
      - ${DB_PORT}:5432
    env_file:
    - .env

  redis:
    container_name: blue_redis
    image: redis:alpine3.17
    restart: always
    env_file:
      - .env
    ports:
      - ${REDIS_PORT}
    expose:
      - ${REDIS_PORT}
    volumes:
      - redis_data:/data



  pgadmin:
    container_name: blue_pgadmin_container
    image: dpage/pgadmin4
    profiles:
      - "dev"
    restart: on-failure
    env_file:
      - ./.env
    volumes:
      - pgadmin_data:/var/lib/pgadmin
    ports:
      - "5050:80"

volumes:
  postgres_data:
  pgadmin_data:
  redis_data:

